components/automate-ui/src/app/entities/layout/layout-sidebar.service.ts

```diff
@@ -1,294 +1,304 @@
-                        icon: 'security',
-                        route: '/settings/policies',
-                        authorized: {
-                            name: 'policies',
-                            allOf: ['/iam/v2/policies', 'get']
-                        },
-                        visible: true
-                    },
-                    {
-                        name: 'Roles',
-                        icon: 'assignment_ind',
-                        route: '/settings/roles',
-                        authorized: {
-                            name: 'roles',
-                            allOf: ['/iam/v2/roles', 'get']
-                        },
-                        visible: true
-                    },
-                    {
-                        name: 'Projects',
-                        icon: 'work',
-                        route: '/settings/projects',
-                        authorized: {
-                            name: 'projects',
-                            allOf: ['/iam/v2/projects', 'get']
-                        },
-                        visible: true
-                    }
-                ],
-                visible: this.isIAMv2$
+    ngOnInit() {
+      this.updateSidebars();
+    }
+
+    ngOnDestroy() {
+      this.isDestroyed.next(true);
+      this.isDestroyed.complete();
+    }
+
+    private updateVisibleValues(sidebars: any): Sidebars {
+      forEach(sidebars, (_value, key) => {
+        const menuGroups = sidebars[key];
+        forEach(menuGroups, (menuGroup) => {
+          if (menuGroup && menuGroup.name) {
+            this.setVisibleValuesToObservables(menuGroup);
+            if (menuGroup.items) {
+              forEach(menuGroup.items, (menuItem) => {
+                this.setVisibleValuesToObservables(menuItem);
+              });
             }
-        ];
+            menuGroup.visible$ = this.setMenuGroupVisibility(menuGroup);
+          }
+        });
+      });
+      return sidebars;
     }
 
-    public getUserProfileSidebar(): MenuItemGroup[] {
-        return [{
-            name: 'User Menu',
-            items: [
-                {
-                    name: 'Profile',
-                    icon: 'person',
-                    route: '.',
-                    visible: true
-                }
-            ],
-            visible: true
-        }];
+    private setVisibleValuesToObservables(item: any): void {
+      if (item.visible$ === undefined) {
+        item.visible$ = new BehaviorSubject(true);
+      } else if (!item.visible$.subscribe) {
+        item.visible$ = new BehaviorSubject(item.visible$);
+      }
+    }
+
+    private setMenuGroupVisibility(menuGroup: any): any {
+      let anyMenuItemsVisible = false;
+      forEach(menuGroup.items, (menuItem) => {
+        if (menuItem.visible$ === undefined) {
+          anyMenuItemsVisible = true;
+        } else if (menuItem.visible$.subscribe) {
+          menuItem.visible$.subscribe((value) => {
```

Comment:
A subscription somewhere other than `ngOnInit` is a red flag to me, so I added a ``console.log(`adding subscription for ${menuItem.name}`);`` inside this subscription, then navigated around the app in the browser. The result: a continuously growing list of subscriptions, so this needs to be fixed:
```
layout-sidebar.service.ts:304 adding subscription for Reports
layout-sidebar.service.ts:304 adding subscription for Scan Jobs
layout-sidebar.service.ts:304 adding subscription for Profiles
layout-sidebar.service.ts:304 adding subscription for Notifications
layout-sidebar.service.ts:304 adding subscription for Data Feeds
layout-sidebar.service.ts:304 adding subscription for Node Integrations
layout-sidebar.service.ts:304 adding subscription for Node Credentials
layout-sidebar.service.ts:304 adding subscription for Node Lifecycle
layout-sidebar.service.ts:304 adding subscription for Users
. . .
```

Reply:
@msorens Updated code